src/tsServer/spawner.ts

```diff
@@ -0,0 +1,154 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+/*
+ * Copyright (C) 2017, 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import API from '../utils/api.js';
+import { ServerType } from './requests.js';
+import { Logger } from '../logger.js';
+import type { TspClientOptions } from '../tsp-client.js';
+import { nodeRequestCancellerFactory } from './cancellation.js';
+import { ITypeScriptServer, ProcessBasedTsServer, TsServerProcessKind } from './server.js';
+import { NodeTsServerProcessFactory } from './serverProcess.js';
+import type { TypeScriptVersion } from './versionProvider.js';
+
+export class TypeScriptServerSpawner {
```

Comment:
Is it maybe planned to have multiple processes for each `TsServerProcessKind`?

```
export const enum TsServerProcessKind {
    Main = 'main',
    Syntax = 'syntax',
    Semantic = 'semantic',
    Diagnostics = 'diagnostics'
}
```

Reply:
Not sure if multiple for each but I don't think so...

It would be one of each at most, I imagine.